assets/app/features/upgrade/controllers/upgrade7-openstack-services.controller.js

```diff
@@ -0,0 +1,99 @@
+    // @ngInject
+    function Upgrade7OpenStackServicesController($translate, openStackFactory) {
+        var vm = this;
+            
+
+        vm.openStackServices = {
+            completed: false,
+            valid: false,
+            checks: {
+                services: {
+                    status: false, 
+                    label: 'upgrade7.steps.openstack-services.codes.services'
+                },
+                backup: {
+                    status: false, 
+                    label: 'upgrade7.steps.openstack-services.codes.backup'
+                }
+            },
+            runOpenStackServices: runOpenStackServices
+        };
+
+
+        /**
+         *  Validate OpenStackServices required for Cloud 7 Upgrade
+         */
+        function runOpenStackServices() {
+   
+            openStackFactory.stopOpenstackServices()
+                .then(
+                    //Success handler. Stop all OpenStackServices successfully:
+                    stopOpenstackServicesSuccess,
+                    //Failure handler:
+                    stopOpenstackServicesError
+                    
+                ).finally(function() {
+                    // Either on sucess or failure, the openStackServices has been completed.
+                    vm.openStackServices.completed = true;
+
+                    // Update openStackServices validity
+                    
+
+                });
+
+            function stopOpenstackServicesSuccess (openStackServicesResponse) {
+
+                vm.openStackServices.checks.services.status = openStackServicesResponse.data.services;
+                vm.openStackServices.valid = vm.openStackServices.checks.services.status;
+                    
+                if (true === vm.openStackServices.checks.services.status) {
+
+                    openStackFactory.createOpenstackBackup()
+                        .then(
+                            //Success handler. Backup OpenStackServices successfully:
+                            createOpenstackBackupSuccess,
+                            //Failure handler:
+                            createOpenstackBackupError
+                        ).finally(function() {
+                            vm.openStackServices.completed = true;
+                            // Update openStackServices validity
+                            vm.openStackServices.valid = vm.openStackServices.checks.backup.status;
+
+                        });
+                } 
+            }
+
+            function stopOpenstackServicesError (errorOpenStackServicesResponse) {
+
+                // Expose the error list to openStackServices object
+                vm.openStackServices.errors = errorOpenStackServicesResponse.data.errors;
+            }
+
+            function createOpenstackBackupSuccess (openStackBackupResponse) {
+                var ResponseData = openStackBackupResponse.data
```

Comment:
start variable names with lowercase letter... or better do as in `createOpenstackBackupError()` and don't add this variable at all.